met-web/src/components/survey/submit/ActionContext.tsx

```diff
@@ -19,14 +21,55 @@ export const ActionContext = createContext<SubmitSurveyContext>({
 export const ActionProvider = ({ children }: { children: JSX.Element }) => {
     const navigate = useNavigate();
     const dispatch = useAppDispatch();
-    const { surveyId } = useParams<SurveyParams>();
+    const isLoggedIn = useAppSelector((state) => state.user.authentication.authenticated);
+    const { surveyId, token } = useParams<SurveyParams>();
     const [savedSurvey, setSavedSurvey] = useState<Survey>(createDefaultSurvey());
     const [isLoading, setIsLoading] = useState(true);
 
     useEffect(() => {
-        loadSurvey();
+        if (!isLoggedIn) {
+            verifyToken();
+        } else {
+            loadSurvey();
+        }
     }, []);
 
+    const verifyToken = async () => {
+        if (!token) {
```

Comment:
just as a thought, maybe involve this part under the try block and throw new Error("message") if !token, catch block will catch it and we can avoid duplication